### ISSUE TITLE
insstall .NET dialog does not work

### ISSUE THREAD

**Issue by alvar-ava** [GPUOpen-Tools/GPU-Reshape#48]
When starting GPU Reshape.exe without the correct .NET installed, there is a dialog asking to install .NET. Pressing Yes or No both result in nothing happening. 
Not sure if it's supposed to open a web page or windows update or similar.

A suggestion would be to list the version of .NET that's needed in the dialog box, then it would be straighforward to find it myself.
![image](https://github.com/GPUOpen-Tools/GPU-Reshape/assets/79447601/903888e6-f37f-497f-8e0a-8928ce24c103)

**Comment by SimpsonGSD** [comment]
What version is required? I can't run it.

**Comment by nixxes-mbouterse** [comment]
Installing .NET 5.0.17 desktop runtime resolved this issue for me.

**Comment by miguel-petersen** [comment]
Hi all,

Please see the /Redist folder in the release package, it currently lists the .NET SDK 5.0 runtimes as a dependency.
https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/sdk-5.0.408-windows-x64-installer

This should definitely be more clear, and the dialogue is insufficient here, particularly as it does not appropriately redirect. I'll see to fixing this.

I'll use this issue going forward to track the problem.

**Comment by miguel-petersen** [comment]
Small update on the above.

So this particular message isn't from GPU Reshape itself, but instead the SDK it was built with. There are a couple of reports out there mentioning the same issue, where the buttons don't actually do anything, unfortunately without easy or relevant fixes. I'll continue searching a bit.

I also have the option of pursuing self-contained builds where the .NET runtimes are embedded within. Quick tests proved a little cumbersome with the current set up, but should be possible nonetheless.

If it was up to me I'd replace the dialog with my own, but I am not sure that is possible. Bootstrapping the app with my own ".NET checker" is a possibility, but far from standard.

For the time being the workaround is to install the .NET SDK linked above. If anyone is unable to launch the application with it installed, please do say so, happy to help.

**Comment by miguel-petersen** [comment]
Branch **issue/48-net-install** now uses .NET self-contained builds. The only big note is that .NET 6.0 was required due to some fixes, however, with it being self-contained it shouldn't matter.

When proved stable, I'll bring it back to main.

**Comment by Hjaltesorgenfrei** [comment]
It is also not listed on <https://gpuopen.com/gpu-reshape/#requirements> as one of the requirements. Which makes it a bit confusing if you come from the website.

**Comment by miguel-petersen** [comment]
That's an great point. Will get it added.

**Comment by miguel-petersen** [comment]
@Hjaltesorgenfrei Both the product page and blog post have been updated.

**Comment by PerfectlyFineCode** [comment]
Avalonia should support NativeAOT, so if possible publishing it as NativeAOT could solve it as there should be no more runtimes needed; although NativeAOT does have a lot of limitations.

**Comment by miguel-petersen** [comment]
@PerfectlyFineCode Yeah NativeAOT is something I considered, however, the limitations get in the way at the moment. Specifically:
- No dynamic loading. All features [resource bounds, etc.] are dynamically loaded plugins)
- No C++/CLI. I make use of /CLI for binding code. Something I might change in the future.

To me, the self contained builds seem like the way forward, though it produces a large amount of DLLs. I could merge them, or pursue single file deployments.

**Comment by miguel-petersen** [comment]
Much needed update to this thread.

The development branch now uses the .NET 8 SDK, which greatly improves that check, and should navigate correctly to the download. Plus a bunch of other changes and new additions. Validated it in a fresh windows installation (VM).

I did make attempts at single file publishing, and embedding the native libraries into it (to avoid installing the desktop runtimes), but I had some issues with some assemblies. I'll shelf that for now, but might bring it back later since it'd make things a bit more tidy.

I'll all be part of an upcoming release, which should be quite soon. 🙂

**Comment by miguel-petersen** [comment]
Hey, the current release features .NET 8, so hopefully this should be resolved. 

I may investigate single file publishing and AOT in the future, but the current setup appears reliable.
